### PR TITLE
[Security] Secvuln 8633 Consul configuration allowed repeated keys

### PR DESCRIPTION
**Pull request by sarahalsmiller**
### Description
 - Due to a known issue in hcl v1, 
 - There is a potential for old unparsed rules to be in the cache, so I added an optional route to replicate the old behavior to maintain backwards compatibility when reading unparsed policies from the cache.

### Testing & Reproduction steps
- New tests pass

### Links
https://github.com/hashicorp/hcl/pull/704 Original HCL PR

### PR Checklist

* [ X] updated test coverage
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [ ] not a security concern

### CODE REVIEW

File: acl/policy.go [hunk 1]
```diff
@@ -338,7 +338,12 @@ func NewPolicyFromSource(rules string, conf *Config, meta *EnterprisePolicyMeta)
 
 	var policy *Policy
 	var err error
-	policy, err = parse(rules, conf, meta)
+	warnOnDuplicateKey := false
+	if conf != nil {
+		warnOnDuplicateKey = conf.WarnOnDuplicateKey
+		fmt.Println(warnOnDuplicateKey)
```
Comment (sarahalsmiller) [inline acl/policy.go:344]:
```suggestion
```

**Review by sophie-gairo — APPROVED** [review]

**Review by dduzgun-security — APPROVED** [review]
LGTM! Awesome work on that

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sarahalsmiller! a backport is missing for this PR [21908] for versions [1.15,1.18,1.19,1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sarahalsmiller! a backport is missing for this PR [21908] for versions [1.15,1.18,1.19,1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sarahalsmiller! a backport is missing for this PR [21908] for versions [1.15,1.18,1.19,1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sarahalsmiller! a backport is missing for this PR [21908] for versions [1.15,1.18,1.19,1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sarahalsmiller! a backport is missing for this PR [21908] for versions [1.15,1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sarahalsmiller! a backport is missing for this PR [21908] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sarahalsmiller! a backport is missing for this PR [21908] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sarahalsmiller! a backport is missing for this PR [21908] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sarahalsmiller! a backport is missing for this PR [21908] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sarahalsmiller! a backport is missing for this PR [21908] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Comment by hc-github-team-consul-core** [comment]
:mega: Hi @sarahalsmiller! a backport is missing for this PR [21908] for versions [1.20] please perform the backport manually and add the following snippet to your backport PR description:
```
<details>
	<summary> Overview of commits </summary>
		- <<backport commit 1>>
		- <<backport commit 2>>
		...
</details>
```

**Pull request merged**